### PR TITLE
Aslts script

### PR DESCRIPTION
**Pull request by SchmErik**
This pull request contains changes to ASLTS test suite script to add binary comparison as well as change slack/norm modes to opt/nopt modes.

### CODE REVIEW

File: tests/aslts/Makefile.def [hunk 1]
```diff
@@ -55,6 +56,7 @@ install_all_modes_of_test_case:	${SETOF_AMLDIRS:%=$(TOP)/tmp/aml/$(aslversion)/%
 		dd=`pwd`; \
 		echo "---- Test path: $$dd" >> $(COMPILER_LOG); \
 		echo "---- Test path: $$dd" >> $(COMPILER_ERROR_LOG); \
+		>&2 echo "Test path: $$dd"; \
```
Comment (zetalog) [inline tests/aslts/Makefile.def]: I'm not sure what the purpose of this line is.
Can "tee" be used instead of?

File: tests/aslts/Makefile.def [hunk 2]
```diff
@@ -47,6 +47,7 @@ empty_all:	FORCE
 ${SETOF_AMLDIRS:%=$(TOP)/tmp/aml/$(aslversion)/%}:
 	@$(INST.dir)
 
+
```
Comment (zetalog) [inline tests/aslts/Makefile.def]: The newline doesn't seem to be correct.

File: tests/aslts/bin/asltsrun [hunk 3]
```diff
@@ -459,8 +460,9 @@ do_summary_of_test_case()
 				out5=$[ 0 + 0x$out5 ]
 			fi
 			memcnt=$[ $memcnt + 1 ]
-		elif [[ "$line" == "Outstanding allocations"* ]]; then
-			outstand1=`echo "$s3" | awk -F"(" '{print $1}'`
+		elif [[ "$line" == *"Outstanding allocations"* ]]; then
+                        trimmedLine=`echo "$line" | awk -F" Outstanding allocations" '{printf $1}'`
```
Comment (zetalog) [inline tests/aslts/bin/asltsrun]: Looks like wrong indentation here.

File: tests/aslts/bin/Do [hunk 4]
```diff
@@ -341,7 +348,12 @@ do_test_cases_make_install()
 
 		if [ $3 != 0 ]; then
 			cd "$dir"
-			make install > /dev/null
+			if [ "$execonly" = "yes" ]; then
```
Comment (zetalog) [inline tests/aslts/bin/Do]: A useful trick in string comparison in shell is:

if [ "x$execonly" = "xyes" ]

The extra "x" ensures no exceptions when environment of execonly doesn't exist.

File: tests/aslts/bin/Do [hunk 5]
```diff
@@ -440,22 +457,22 @@ make_install_3()
...
+	local lexem list nparam=$2 action=100 execonly=$4
 
 	lexem=`echo "$3" | awk '{ print $2}'`
 
 	if [ $nparam -le 1 ]; then
 		usage
 		do_exit 1 "Bad parameters 0"
 	elif [ $lexem == ASLTS -o $lexem == aslts ]; then
-		if [ $nparam != 2 ]; then
+		if [ $nparam != 3 ]; then
```
Comment (zetalog) [inline tests/aslts/bin/Do]: Should the failure hints be corrected accordingly?

File: tests/aslts/bin/Do [hunk 6]
```diff
@@ -467,15 +484,18 @@ run_asl_compiler()
 		list=`echo "$3" | cut -c 3-`
 	fi
 
+	list=`echo $list | rev | cut -c 4- | rev ` # cut the last item from the list. It's a yes or no from the diasm option.
```
Comment (zetalog) [inline tests/aslts/bin/Do]: Too long comments, you may make it a separate line.
And what does the diasm mean?

File: tests/aslts.sh [hunk 7]
```diff
@@ -114,13 +115,13 @@ run_aslts() {
 
 	if [ "x$TEST_CASES" = "x" ]; then
 		# Compile all ASL test modules
-		Do 0 aslts
+		Do 0 aslts $EXECONLY
```
Comment (zetalog) [inline tests/aslts.sh]: Can the first argument of "Do" cover this option?
For example "Do 9 aslts"?

File: tests/aslts/bin/Do (hunk 4 above)
Comment (SchmErik) [inline tests/aslts/bin/Do]: This is a pretty neat trick. Thanks!

**Pull request merged**